lib/srv/db/postgres/engine.go

```diff
@@ -362,3 +387,52 @@ func (e *Engine) getConnectConfig(ctx context.Context, sessionCtx *common.Sessio
 	}
 	return config, nil
 }
+
+// getBindParameters converts prepared statement parameters from the Postgres
+// wire protocol Bind message into their string representations for including
+// in the audit log.
+func getBindParameters(msg *pgproto3.Bind) (parameters []string) {
+	// Each parameter can be either a text or a binary which is determined
+	// by "parameter format codes" in the Bind message (0 - text, 1 - binary).
+	//
+	// Be a bit paranoid and make sure that number of format codes matches the
+	// number of parameters, or there are no format codes in which case all
+	// parameters will be text.
+	if len(msg.ParameterFormatCodes) != 0 && len(msg.ParameterFormatCodes) != len(msg.Parameters) {
+		logrus.Warnf("Postgres parameter format codes and parameters don't match: %#v.", msg)
+		return parameters
+	}
+	for i, p := range msg.Parameters {
+		// According to Bind message documentation, if there are no parameter
+		// format codes, it may mean that either there are no parameters, or
+		// that all parameters use default text format.
+		if len(msg.ParameterFormatCodes) == 0 || msg.ParameterFormatCodes[i] == parameterFormatCodeText {
+			// Text parameters can just be converted to their string
+			// representation.
+			parameters = append(parameters, string(p))
+		} else if msg.ParameterFormatCodes[i] == parameterFormatCodeBinary {
+			// For binary parameters, just put a placeholder to avoid
+			// spamming the audit log with unreadable info.
+			parameters = append(parameters, "<binary>")
+		} else {
+			// Should never happen but...
+			logrus.Warnf("Unknown Postgres parameter format code: %#v.", msg)
+			parameters = append(parameters, "<unknown>")
+		}
```

Comment:
```suggestion
		if len(msg.ParameterFormatCodes) == 0 {
			parameters = append(parameters, string(p))
			continue
		}
		switch msg.ParameterFormatCodes[i] {
			case parameterFormatCodeText:
				// Text parameters can just be converted to their string
				// representation.
				parameters = append(parameters, string(p))
			case parameterFormatCodeBinary:
				// For binary parameters, just put a placeholder to avoid
				// spamming the audit log with unreadable info.
				parameters = append(parameters, "<binary>")
			default:
				// Should never happen but...
				logrus.Warnf("Unknown Postgres parameter format code: %#v.", msg)
				parameters = append(parameters, "<unknown>")
		}
```